src/setfit/onnx.py

```diff
@@ -0,0 +1,227 @@
+from .modeling import SetFitModel
+from sentence_transformers import models
+import numpy as np
+import onnx
+import warnings
+from typing import Optional, Union
+import torch
+from sklearn.linear_model import LogisticRegression
+
+
+class ONNXSetFitModel(torch.nn.Module):
+    def __init__(
+        self,
+        model_body,
+        pooler,
+        model_head: Optional[Union[torch.nn.Module, LogisticRegression]] = None,
+    ):
+        super().__init__()
+
+        self.model_body = model_body
+        self.pooler = pooler
+        self.model_head = model_head
+
+    def forward(self, input_ids, attention_mask, token_type_ids):
+        hidden = self.model_body(input_ids, attention_mask, token_type_ids)
+        hidden = {"token_embeddings": hidden[0], "attention_mask": attention_mask}
+
+        hidden = self.pooler(hidden)["sentence_embedding"]
+        if self.model_head is None:
+            return hidden
+
+        out = self.model_head(hidden)
```

Comment:
Would this be more descriptive or does it only apply to the PyTorch model?
```suggestion
        logits = self.model_head(hidden)
```